### ISSUE TITLE
Interface for "components that need curation"

### ISSUE THREAD

**Issue by fossygirl** [clearlydefined/website#309]
One way we can give folks a way to get going on curations is to provide a page on the website with random components that are missing declared licenses. We could then invite them to easily curate one of those components right there. It's hard to know where to start right now on which components need curations, so let's give folks a way in and pick out components we know need curations and display them to the user.

**Comment by jeffmcaffer** [comment]
Can this be covered by sorting by score? Things that have a low score need curation. Is that enough?

**Comment by fossygirl** [comment]
@jeffmcaffer Yes, I think so. We just need a UI that surfaces these for users to pick from and filter and then work on.

**Comment by jeffmcaffer** [comment]
do you think the current sort by score and filter is enough?

**Comment by fossygirl** [comment]
@jeffmcaffer No, I don't, since we're going to have people coming to the interface who have no components in mind they're searching for or interested in from another UI. We want to "feed" them new things that need to be curated, so surface them and then let them filter from there once they have a queue to work from.

**Comment by jeffmcaffer** [comment]
ah, I see, you are suggesting a landing page of sorts that pre-populates a list of high value, low scoring components as a means of getting people engaged who don't already have a set of things in which they are interested. With the new database stuff we should be able to do a "high impact" page where the user can filter quite coarsely (e.g., npm vs maven) and then we filter and find the ones with the lowest scores. 

Would be great to have popularity in there as well. we'd have to pull that data from time to time. Likely will be doing that anyway to support some of the progress reporting etc that we want to do.

**Comment by fossygirl** [comment]
We want this page to be a browse page that displays all definitions.

*We need sorting, filtering, and browsing functionality on this page.
*The default view for a person coming to the page for the first time should be definitions with a score below 70.

**Comment by storrisi** [comment]
@fossygirl @jeffmcaffer  Just to recap and make clear all the requirements:

* a new page "Browse" should be created
* The page will contain a list of definitions, returned from a specific API, and a selection of all the available providers, which allow the user only retrieve data by specific provider (e.g. npm, pypi, etc.)
* The API should accept the provider as a param, and return a list of some definitions with score < 70, sorted by default by popularity, as @jeffmcaffer mentioned in a previous comment
* The definitions list in this page shouldn't have any of the functionality of the Definition Page, if the user need to make some kind of activities on the definitions (e.g. curate) he can go to the details of a specific definition
* The user can filter and sorting for all the data of definition, and by popularity

If I am not wrong, one of the goals is to display stats onto the browse page, is this something we should implement (and define) as part of this issue?

**Comment by fossygirl** [comment]
@storrisi A couple things:

1. No, we do not want stats on this page. We decided yesterday that's a new page: https://github.com/clearlydefined/website/issues/358

2. I don't know that it's fair to say "the definitions list in this page shouldn't have any of the functionality of the Definition page." We still want the same sorting and filtering capability. And we also want edits the user does want to do from the Browse page to be very easy with very few clicks. Whether this means taking them to the Definition page or implementing Definition page functionality on the Browse page I think we should have @jeffmcaffer weigh in on.

**Comment by storrisi** [comment]
@jeffmcaffer do you have any additional comments on this?

If, as supposed by @fossygirl , much of the functionalities would be shared with the Definitions List, i would like to consider a refactor of the PageDefinition component while working on this, as well for what you've done with the drop functionality.

**Comment by jeffmcaffer** [comment]
Summarizing the chat we had in our planning meeting... I agree with @fossygirl. The Browse page should be more or less like the Definitions page but 
* be pre-populated with definitions we have the fit the given sort/filter criteria (whereas the Definitions page has only things the user has added)
* default to sort by lowest score first

It is important for ClearlyDefined as a whole that if a user is looking at some data and they see a problem, they are able to fix it, regardless (more or less) of where they are.
